lib/crypto/packaged.py

```diff
@@ -153,7 +151,8 @@ def sign_multi(file_path, endpoint, guid):
             xpi_info = parse_xpi(xpi, check=False)
             if xpi_info['type'] == amo.ADDON_EXTENSION:
                 # Sign the internal extensions in place.
-                cert_serial_nums.append(call_signing(xpi, endpoint, guid))
+                cert_serial_nums.append(
+                    call_signing(xpi, endpoint, xpi_info['guid']))
```

Comment:
This is a huge security hole. We can't sign inner XPIs with certs for GUIDs that the uploading developer doesn't own the listing for. Coming up with a proper solution will require some thought, but we definitely can't do this.